src/vsphere_cpi/spec/integration/attach_tag_to_vm_spec.rb

```diff
@@ -0,0 +1,313 @@
+      tag_id_info = tagging_tag_api.create(cis_tagging_tag_create)
+      tag_id_info.value
+    end
+
+    def create_cat_and_tag
+      cate_config_1 = {
+        "create_spec" => {
+          "name" => "category-name-a",
+          "description" => "category contains multi-tag, associable to vm, multi-cardinality",
+          "associable_types" => [],
+          "cardinality" => "MULTIPLE"
+        }
+      }
+      cate_config_2 = {
+          "create_spec" => {
+            "name" => "category-name-b",
+            "description" => "category contains multi-tag, not associable to vm, multi-cardinality",
+            "associable_types" => ["VirtualApp", "StoragePod", "Datacenter"],
+            "cardinality" => "MULTIPLE"
+          }
+      }
+      cate_config_3 = {
+        "create_spec" => {
+          "name" => "category-name-c",
+          "description" => "category contains multi-tag, associable to vm, single-cardinality",
+          "associable_types" => [],
+          "cardinality" => "SINGLE"
+        }
+      }
+
+      cate_config_array = [cate_config_1, cate_config_2, cate_config_3]
+      cate_config_array.each do |cate_config|
+        cate_id = create_category(cate_config)
+        cate_ids << cate_id unless cate_id.nil?
+      end
+
+      tag_config_array = Array.new
+      tag_config_array << { "create_spec" => { "name" => "tag-name-a-1", "description" => "null", "category_id" => cate_ids[0] } }
+      tag_config_array << { "create_spec" => { "name" => "tag-name-a-2", "description" => "null", "category_id" => cate_ids[0] } }
+      tag_config_array << { "create_spec" => { "name" => "tag-name-a-3", "description" => "null", "category_id" => cate_ids[0] } }
+      tag_config_array << { "create_spec" => { "name" => "tag-name-b-1", "description" => "null", "category_id" => cate_ids[1] } }
+      tag_config_array << { "create_spec" => { "name" => "tag-name-b-2", "description" => "null", "category_id" => cate_ids[1] } }
+      tag_config_array << { "create_spec" => { "name" => "tag-name-c-1", "description" => "null", "category_id" => cate_ids[2] } }
+      tag_config_array << { "create_spec" => { "name" => "tag-name-c-2", "description" => "null", "category_id" => cate_ids[2] } }
+      tag_config_array << { "create_spec" => { "name" => "tag-name-c-3", "description" => "null", "category_id" => cate_ids[2] } }
+
+      tag_config_array.each do |tag_config|
+        unless tag_config["create_spec"]["category_id"].nil?
+          tag_id = create_tag(tag_config)
+          tag_ids << tag_id unless tag_id.nil?
+        end
+      end
+    end
+
+    def delete_category
+      tag_ids.each do |tag_id|
+        tagging_tag_api.delete(tag_id)
+      end
+    end
+
+    def delete_tag
+      cate_ids.each do |cate_id|
+          tagging_category_api.delete(cate_id)
+      end
+    end
+
+    def verify_tags(vm_mob_id, attached_tags)
+      tag_assoc_api = VSphereAutomation::CIS::TaggingTagAssociationApi.new(tag_client)
```

Comment:
Make this a private function for fututre reuse.

Reply:
Done.